shell/platform/android/io/flutter/view/AccessibilityBridge.java

```diff
@@ -266,14 +297,29 @@ public AccessibilityNodeInfo createAccessibilityNodeInfo(int virtualViewId) {
         if (object.hasAction(Action.SCROLL_LEFT) || object.hasAction(Action.SCROLL_UP)
                 || object.hasAction(Action.SCROLL_RIGHT) || object.hasAction(Action.SCROLL_DOWN)) {
             result.setScrollable(true);
+
             // This tells Android's a11y to send scroll events when reaching the end of
             // the visible viewport of a scrollable, unless the node itself does not
             // allow implicit scrolling - then we leave the className as view.View.
+            //
+            // We should prefer setCollectionInfo to the class names, as this way we get "In List"
+            // and "Out of list" announcements.  But we don't always know the counts, so we
+            // can fallback to the generic scroll view class names.
+            // TODO(dnfield): We should add semantics properties for rows and columns in 2 dimensional lists, e.g.
+            // GridView.  Right now, we're only supporting ListViews and only if they have scroll children.
             if (object.hasFlag(Flag.HAS_IMPLICIT_SCROLLING)) {
                 if (object.hasAction(Action.SCROLL_LEFT) || object.hasAction(Action.SCROLL_RIGHT)) {
-                    result.setClassName("android.widget.HorizontalScrollView");
+                    if (shouldSetCollectionInfo(object)) {
+                        result.setCollectionInfo(AccessibilityNodeInfo.CollectionInfo.obtain(0, object.scrollChildren, false));
```

Comment:
What does the `0` do here? columns?

Reply:
Rows - this one is horizontal, so we're setting columns to the number of children and leaving rows as 0.

Reply:
Ahhh, makes sense

Reply:
Want me to add inline comments for that?

Reply:
Wouldn't hurt